### PR TITLE
Fix sandwich attacks when adding liquidity into unbalanced pools

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

File: contracts/periphery/DMMRouter02.sol [hunk 1]
```diff
@@ -165,6 +179,9 @@ contract DMMRouter02 is IDMMRouter02 {
         if (pool == address(0)) {
             pool = IDMMFactory(factory).createPool(tokenA, tokenB, ampBps);
         }
+        // if we add liquidity to an existing pools this is unamplifed pool
```
Comment (Anyhowclick) [inline contracts/periphery/DMMRouter02.sol]: if we add liquidity to an existing pool this is unamplified pool

File: contracts/periphery/DMMRouter02.sol [hunk 2]
```diff
@@ -203,12 +221,16 @@ contract DMMRouter02 is IDMMRouter02 {
         if (pool == address(0)) {
             pool = IDMMFactory(factory).createPool(token, weth, ampBps);
         }
+        // if we add liquidity to an existing pools this is unamplifed pool
```
Comment (Anyhowclick) [inline contracts/periphery/DMMRouter02.sol]: edit as above

**Review by Anyhowclick — CHANGES_REQUESTED** [review]

**Review by ducquangkstn — APPROVED** [review]

**Review by ducquangkstn — APPROVED** [review]

File: contracts/periphery/DMMRouter02.sol (hunk 1 above)
Comment (ducquangkstn) [reply contracts/periphery/DMMRouter02.sol]: updated

File: contracts/periphery/DMMRouter02.sol (hunk 2 above)
Comment (ducquangkstn) [reply contracts/periphery/DMMRouter02.sol]: updated

File: contracts/periphery/DMMRouter02.sol [hunk 3]
```diff
@@ -165,6 +179,9 @@ contract DMMRouter02 is IDMMRouter02 {
         if (pool == address(0)) {
             pool = IDMMFactory(factory).createPool(tokenA, tokenB, ampBps);
         }
+        // if we add liquidity to an existing pool, this is an unamplifed pool
```
Comment (Anyhowclick) [inline contracts/periphery/DMMRouter02.sol:182]: unamplified, missing "i"

**Review by Anyhowclick — CHANGES_REQUESTED** [review]

**Pull request merged**